internal/storages/builder/builder.go

```diff
@@ -27,9 +28,10 @@ import (
 func GetStorage(ctx context.Context, stCfg *domains.StorageConfig, logCgf *domains.LogConfig) (
 	storages.Storager, error,
 ) {
-	if stCfg.Directory != nil {
+	envCfg := os.Getenv("STORAGE_TYPE")
```

Comment:
It is better to implement the configuration in the next way:

```
storage:
  type: "s3"

   s3_config: 
     bucket: "test"
  
   directory_config: 
     path: "/tmp"
```

We should introduce the `storage.type` parameter and depending on the type provided you can use either S3 config or Directory.

Keep in mind that the env vars mapping should be done via viper/cobra usage [here](https://github.com/GreenmaskIO/greenmask/blob/main/cmd/greenmask/cmd/root.go)

Reply:
I think it is important now, because `AutomaticEnv` does not work when we provide the variable `STORAGE_DIRECTORY_PATH=/tmp`. The default config file assembles the S3 storage (not the Directory). It would be fine to introduce the next keys in the config:

- storage.type
- storage.s3
- storage.directory

Then the variables will be dynamically working using the next variable naming:
* For path setting in Directory `STORAGE_DIRECTORY_PATH=/tmp`
* For bucket setting in S3 `STORAGE_S3_BUCKET=testbucket`